src/lib/upload.ts

```diff
@@ -0,0 +1,140 @@
+export class Config {
+  static repo: string = 'internetofwater/geoconnex.us'
+  // static repo: string = 'webb-ben/geoconnex.us'
+  static baseBranch: string = 'master'
+}
+
+async function createBranch(
+  headers: HeadersInit,
+  repo: string,
+  baseBranch: string,
+  branchName?: string
+) {
+  const baseBranchUrl = `https://api.github.com/repos/${repo}/git/refs/heads/${baseBranch}`
+  const baseBranchData = await fetch(baseBranchUrl, { headers }).then((res) => res.json())
+  const newBranchUrl = `https://api.github.com/repos/${repo}/git/refs`
+  console.log(baseBranchData)
+  const newBranchData = {
+    ref: `refs/heads/${branchName}`,
+    sha: baseBranchData.object.sha
+  }
+
+  const newBranchResponse = await fetch(newBranchUrl, {
+    method: 'POST',
+    headers,
+    body: JSON.stringify(newBranchData)
+  })
+
+  if (!newBranchResponse.ok) {
+    throw new Error('Error creating new branch.')
+  }
+}
+
+export async function ensureContentIsNew(
+  headers: HeadersInit,
+  repo: string,
+  baseBranch: string,
+  remoteFilePath: string,
+  content: string
+) {
+  /* Check if the file already exists, and if it does, check if the content is the same. If so, throw an error*/
+
+  const fileUrl = `https://api.github.com/repos/${repo}/contents/${remoteFilePath}?ref=${baseBranch}`
+
+  const fileResponse = await fetch(fileUrl, {
+    headers
+  })
+
+  // We expect it to be a 404 when uploading a new file but other error codes should be thrown
+  if (!fileResponse.ok && fileResponse.status !== 404) {
+    throw new Error(`Failed to get file: ${fileResponse.status} ${fileResponse.statusText}`)
+  }
+
+  // If the file already exists, throw an error if the content is also the same
+  const fileData = await fileResponse.json()
+  if (fileResponse.status !== 404) {
+    const existingFileContent = atob(fileData.content.replace(/\n/g, ''))
+    console.log(existingFileContent, content)
+    if (existingFileContent === content) {
+      throw new Error('No changes detected; skipping pull request.')
+    } else {
+      console.log('Content has changed; creating pull request.')
+    }
+  }
+}
```

Comment:
This function doesn't seem to be working but can't figure out why. Did this work previously with you? Worried it might be something related to OS/encoding specifics.